models/maat/R/model2netcdf.MAAT.R

```diff
@@ -37,72 +45,66 @@
-  ###end <- strftime(as.Date(end_date,origin="1970-01-01"), "%Y-%m-%d")
-  ###start <- as.Date(start_date,"%Y-%m-%d")
-  ###end <- as.Date(end_date,"%Y-%m-%d")
-  #num.years <- length(unique(year))
-  #years <- unique(year)
-  #timestep.s <- 86400
-  # !!
-  
+
   ### Determine number of years and output timestep
-  # !!! Need to output year and day in out.csv to be able to determine timestep.
   days <- as.Date(start_date):as.Date(end_date)
   year <- strftime(as.Date(days,origin="1970-01-01"), "%Y")
   num.years <- length(unique(year))
-  years <- unique(year)
-  timestep.s <- 86400 / 48 # hard-coded for Ameriflux....need to change, read from met file?
-  
+  maat.dates <- as.Date(maat.output$time, format = "%m/%d/%y")
+  dims <- dim(subset(maat.output, as.Date(time, format = "%m/%d/%y") == seq(as.Date(start_date),
+                                                                            by='days',length=1)))
+  timestep.s <- 86400 / dims[1]
+
   ### Setup outputs for netCDF file in appropriate units
-  for (y in years){
+  for (y in unique(year)){
     if (file.exists(file.path(outdir, paste(y,"nc", sep=".")))) {
       next ## skip, model output already present.
     }
     
     print(paste("---- Processing year: ", y))  # turn on for debugging
-    output <- list()
-    
-    ## Subset data for processing - !!! NEED TO ADD YEAR TO OUTPUT FILE TO SUBSET OVER MULT YEARS
-    #sub.maat.output <- subset(maat.output, year == y)
-    #sub.maat.output.dims <- dim(sub.maat.output)
-    #dayfrac = 1 / out.day
-    #step <- seq(0, 0.99, 1 / out.day)
-    
-    
     
+    ## Subset data for processing
+    sub.maat.output <- subset(maat.output, format(maat.dates, "%Y") == y)
+    sub.maat.dates <- as.Date(sub.maat.output$time, format = "%m/%d/%y")
+    sub.maat.doy <- yday(sub.maat.dates)
+    sub.maat.output.dims <- dim(sub.maat.output)
+    dayfrac <- 1 / dims[1]
+    day.steps <- seq(0, 0.99, 1 / dims[1])
+
     ### standard variables: Carbon Pools [not currently relevant to MAAT]
-    output[[1]] <- y   # Year - REMOVE!  ALREADY IN THE NETCDF AS TIME!!
-    output[[2]] <- (maat.output$A)    # assimilation in umolsC/m2/s - OR KEEP AS ASSIMILATION?
-    #output[[2]] <- c(maat.output$A,-999)
+    output <- list()                              # create empty output
+    out.year <- as.numeric(rep(y,sub.maat.output.dims[1]))
+    output[[1]] <- out.year                       # Simulation year
+    output[[2]] <- sub.maat.doy+day.steps         # Fractional day - NEED TO IMPLEMENT  
+    output[[3]] <- (sub.maat.output$A)            # assimilation in umolsC/m2/s
+    
+    ## TODO: ADD MORE MAAT OUTPUTS HERE ##
     
     #******************** Declare netCDF variables ********************#
-    ## TODO !!!THIS BIT NEEDS UPDATING TO CAPTURE HIGH-FREQUENCY (sub daily) OUTPUTS !!!
+    ## This version doesn't provide enough output timesteps when running with met data that has
+    ## a step greater than 1 per day
```

Comment:
You might want to throw an error or warning explicitly if you detect a timestep you can't process.  Hope to have all temporal downscaling handled BEFORE met2model functions using common tools, so you shouldn't ever have to implement that downscaling internally.